src/layer/GeoJSON.js

```diff
@@ -235,7 +235,7 @@ export function geometryToLayer(geojson, options) {
 function _pointToLayer(pointToLayerFn, geojson, latlng, options) {
 	return pointToLayerFn ?
 		pointToLayerFn(geojson, latlng) :
-		new Marker(latlng, options && options.markersInheritOptions && options);
+		new Marker(latlng, options && options.markersInheritOptions);

```

Comment:
the return value is now not `options` anymore, this needs more investigations

Reply:
The first `options &&` ensures that `options` is an non-undefined object, whereas the second `&& options` handles the semantics of `markersInheritOptions` (returning `options` if `markersInheritOptions` is truthy)

Should be `options?.markersInheritOptions && options` instead, or maybe `options?.markersInheritsOptions ? options : {}`. The behaviour is the same, but the fancy syntax makes the purpose a bit clearer IMO.

Reply:
I prefer `options?.markersInheritOptions && options`